### PR TITLE
Add new unique faker distribution

### PR DESCRIPTION
**Pull request by qubixes**
I added a new mixin class for creating new unique distributions more easily.

### CODE REVIEW

File: metasynth/distribution/base.py [hunk 1]
```diff
@@ -254,3 +254,28 @@ def information_criterion(self, values):
...
+        while n_retry < 1e5:
+            new_val = super().draw()
+            if new_val not in self.key_set:
+                self.key_set.add(new_val)
+                return new_val
+            n_retry += 1
+        raise ValueError(f"Failed to draw unique string after {n_retry} tries.")
+
+    def information_criterion(self, values):
+        return 9999999
```
Comment (vankesteren) [inline metasynth/distribution/base.py:281]: This is very elegant, we can use this for the other unique distributions too no?

File: metasynth/distribution/faker.py [hunk 2]
```diff
@@ -59,3 +59,8 @@ def _param_schema(cls):
             "faker_type": {"type": "string"},
             "locale": {"type": "string"},
         }
+
+
+@metadist(implements="core.unique_faker", var_type="string")
```
Comment (vankesteren) [inline metasynth/distribution/faker.py:64]: Would it also work if we just do 
`@metadist(implements="core.unique_faker", var_type="string", is_unique="true")`
without subclassing `UniqueMixin`?

Would it make sense to call `UniqueMixin` something more descriptive about what method it overloads? Such as `UniqueSampler`, `UniqueDraws`, or even `UniqueRejectionSampler`?

**Review by vankesteren — APPROVED** [review]

File: metasynth/distribution/base.py (hunk 1 above)
Comment (qubixes) [reply metasynth/distribution/base.py:281]: Yes, that is indeed the point!

File: metasynth/distribution/faker.py (hunk 2 above)
Comment (qubixes) [reply metasynth/distribution/faker.py:64]: The term "Mixin" is the kind of class this is, so I think that will help people familiar with the concept (or can find it on google). We can also call it `UniqueDistributionMixin` or `UniqueSamplerMixin`, etc. That might be the most descriptive. It's basically a copy from the `UniqueRegexDistribution`, but I don't want to mess with that right now, while I'm messing with the regex distributions...

It is probably possible to implement a decorator like that:

https://stackoverflow.com/questions/38405619/how-to-mixin-behavior-using-class-decorators-in-python

But honestly, I think that will just drive up the complexity a little too much..

Comment (vankesteren) [reply metasynth/distribution/faker.py:64]: Agreed with everything. Either of those names you suggested are good!

**Comment by qubixes** [comment]
@vankesteren Should we already merge or should we work on documenting the feature?

**Comment by vankesteren** [comment]
> work on documenting the feature?

What do you mean? As in: developer documentation?

**Comment by qubixes** [comment]
Yes, I suppose mostly developer documentation.

**Comment by qubixes** [comment]
But maybe we should organize that in one go instead.

**Comment by vankesteren** [comment]
Yeah, let's merge!

**Pull request merged**